src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -153,18 +186,23 @@ export class WorkspaceSavedObjectsClientWrapper {
       id: string,
       options: SavedObjectsDeleteOptions = {}
     ) => {
-      if (this.isRelatedToWorkspace(type)) {
-        await this.validateSingleWorkspacePermissions(id, wrapperOptions.request, [
-          WorkspacePermissionMode.Management,
-        ]);
-      }
-
       const objectToDeleted = await wrapperOptions.client.get(type, id, options);
-      await this.validateMultiWorkspacesPermissions(
-        objectToDeleted.workspaces,
+      const workspacePermitted = await this.validateMultiWorkspacesPermissions(
+        objectToDeleted.workspaces!,
         wrapperOptions.request,
-        WorkspacePermissionMode.Management
+        [WorkspacePermissionMode.Management, WorkspacePermissionMode.LibraryWrite]
       );
+
+      if (!workspacePermitted) {
+        const objectsPermitted = await this.validateMultiObjectsPermissions(
+          [{ type, id }],
+          wrapperOptions.request,
+          [WorkspacePermissionMode.Management, WorkspacePermissionMode.Write]
```

Comment:
Do we need to check `WorkspacePermissionMode.LibraryWrite` when type equal WORKSPACE?